geo/src/algorithm/minimum_rotated_rect.rs

```diff
@@ -0,0 +1,103 @@
+use num_traits::Float;
+
+use crate::{
+    algorithm::{centroid::Centroid, rotate::Rotate, BoundingRect, CoordsIter},
+    Area, ConvexHull, CoordFloat, GeoFloat, GeoNum, LinesIter, Polygon,
+};
+/// Return the minimum bounding rectangle(MBR) of geometry
+/// reference: https://en.wikipedia.org/wiki/Minimum_bounding_box
+/// minimum rotated rect is the rectangle that can enclose all points given
+/// and have smallest area of all enclosing rectangles
+/// the rect can be any-oriented, not only axis-aligned.
+///
+/// # Examples
+///
+/// ```
+/// use geo_types::{line_string, polygon, LineString, Polygon};
+/// use geo::MinimumRotatedRect;
+/// fn returns_polygon_mbr() {
```

Comment:
This defines a doc-test, but the test doesn't actually run anything, it only defines the function. 

if you remove the function declaration, and just have the inner code, a `main` will be synthesized and the test will run.

Reply:
fixed